src/Juvix/Core/FromFrontend.hs

```diff
@@ -162,71 +249,83 @@ transformTermHR (FE.Primitive (FE.Prim p)) = do
   where
     primTy param p = HR.PrimTy <$> HM.lookup p (P.builtinTypes param)
     primVal param p = HR.Prim <$> HM.lookup p (P.builtinValues param)
-transformTermHR (FE.List (FE.ListLit es)) =
-  makeList <$> traverse transformTermHR es
+transformTermHR (FE.List l) =
+  throwFF $ ListUnimplemented l
 transformTermHR (FE.Tuple (FE.TupleLit es)) =
   makeTuple <$> traverse transformTermHR es
 transformTermHR (FE.Block (FE.Bloc e)) =
   transformTermHR e
-transformTermHR (FE.ExpRecord (FE.ExpressionRecord fs)) =
-  makeRecord <$> traverse traverseField (toList fs)
-  where
-    traverseField (FE.NonPunned n e) = (n,) <$> transformTermHR e
+transformTermHR (FE.ExpRecord r) =
+  throwFF $ ExpRecordUnimplemented r
 transformTermHR (FE.ArrowE a) = transformArrow a
 transformTermHR (FE.NamedTypeE (FE.NamedType' _ e)) =
   -- TODO the name will only become relevant (outside of arrows)
   -- when refinements are supported
   transformTermHR e
 transformTermHR (FE.RefinedE r) =
-  throwFF $ RefinementsUnsupported r
+  throwFF $ RefinementsUnimplemented r
 transformTermHR (FE.UniverseName i) =
   -- TODO for universe polymorphism
-  throwFF $ UniversesUnsupported i
+  throwFF $ UniversesUnimplemented i
 transformTermHR (FE.Parened e) = transformTermHR e
 
+transformSimpleLet :: FE.Let -> Env primTy primVal (HR.Term primTy primVal)
+transformSimpleLet e@(FE.LetGroup name (clause :| []) body) = do
+  let FE.Like args cbody = clause
+  rhs <-
+    toElim (FE.Let e)
+      =<< foldr HR.Lam <$> transformTermHR cbody <*> traverse isVarArg args
+  HR.Let Usage.Omega (NameSymbol.fromSymbol name) rhs <$> transformTermHR body
```

Comment:
This seems reasonable modulo the usage question.

The frontend usage is my bad, given we don't have let sig rolled up I don't think

Reply:
Yeah I guess we'll need an explicit usage here for now, that's fine.

Reply:
Split to new issue https://github.com/metastatedev/juvix/issues/581